### DISCUSSION TITLE
TCA navigation and in line coordinator pattern

### DISCUSSION THREAD

**Discussion by importRyan** [pointfreeco/swift-composable-architecture#2286] [Q&A]
With John Patrick Morgan's TCA Coordinators, coordinator reducers can present screens or other in-line coordinators who present other screens in the same linear path. This enabled clean modularization. For example, the 2FA coordinator could imported by onboarding and logged out modules and inserted in-line with little knowledge about number of screens etc.

In the new TCA navigation API, is such composition possible? Can NavigationStackStore state be scoped with the same ergonomics? 

Catching up on where this landed, it [looks](https://github.com/pointfreeco/swift-composable-architecture/discussions/2149) like a similar pattern can work, but there's, at first blush, no built-in-boiler-avoidance for helping the "child" coordinator add and remove from the navigation path owned by its parent.

**Comment by mbrandonw** [comment]
TCA's navigation tools are built to mimic exactly what vanilla SwiftUI gives you, except allowing each feature presented to be powered by a `Store`. So, whatever is possible with vanilla SwiftUI should still be possible in TCA. And similarly, if there is something lacking in vanilla SwiftUI that causes you to build tooling for it, then the same may need to be done in TCA too.

Having said that, if you can provide a concrete example of what you are talking about I could better answer whether it is possible in TCA's navigation tools. There are many ways to interpret your hypothetical 2FA example, and so I want to make sure we are talking about the same thing.
